### PR TITLE
Speed up git destination with mostly excluded files

### PR DESCRIPTION
**Pull request by aaron-skydio**

### CODE REVIEW

File: java/com/google/copybara/git/AddExcludedFilesToIndex.java [hunk 1]
```diff
@@ -68,13 +70,18 @@ void findSubmodules(Console console) throws RepoException {
   /**
    * Adds all the excluded files and submodules.
    */
-  void add() throws RepoException, IOException {
+  void add(Console console) throws RepoException, IOException {
```
Comment (danny-skydio) [inline java/com/google/copybara/git/AddExcludedFilesToIndex.java:73]: is passing the console to methods a pattern in copybara?

File: java/com/google/copybara/git/AddExcludedFilesToIndex.java [hunk 2]
```diff
@@ -89,17 +96,111 @@ void add() throws RepoException, IOException {
...
+    private final String filename;
+    private boolean included;
+
+    private PathTreeLeaf(String filename, boolean included) {
+      this.filename = filename;
+      this.included = included;
+    }
+  }
+
+  private static final class PathTree {
```
Comment (danny-skydio) [inline java/com/google/copybara/git/AddExcludedFilesToIndex.java]: Could you add some comments to this class?

**Review by danny-skydio — APPROVED** [review]

File: java/com/google/copybara/git/AddExcludedFilesToIndex.java (hunk 2 above)
Comment (aaron-skydio) [reply java/com/google/copybara/git/AddExcludedFilesToIndex.java]: Done

File: java/com/google/copybara/git/AddExcludedFilesToIndex.java (hunk 1 above)
Comment (aaron-skydio) [reply java/com/google/copybara/git/AddExcludedFilesToIndex.java:73]: Yeah it seems like that's how they do things

**Pull request merged**